cli/cmd/common/diff_test.go

```diff
@@ -0,0 +1,142 @@
+package common
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDiff(t *testing.T) {
+	type args struct {
+		a map[string]interface{}
+		b map[string]interface{}
+	}
+
+	tests := []struct {
+		name     string
+		args     args
+		expected string
+	}{
+		{
+			name: "Two empty maps should return an empty string",
+			args: args{
+				a: map[string]interface{}{},
+				b: map[string]interface{}{},
+			},
+			expected: "",
+		},
+		{
+			name: "Two equal maps should return the map parsed as YAML",
+			args: args{
+				a: map[string]interface{}{
+					"foo": "bar",
+					"baz": "qux",
+					"liz": map[string]interface{}{"qux": []string{"quux", "quuz"}},
+				},
+				b: map[string]interface{}{
+					"foo": "bar",
+					"baz": "qux",
+					"liz": map[string]interface{}{"qux": []string{"quux", "quuz"}},
+				},
+			},
+			expected: "  baz: qux\n  foo: bar\n  liz:\n    qux:\n    - quux\n    - quuz\n",
+		},
+		{
+			name: "New elements should be prefixed with a plus sign",
+			args: args{
+				a: map[string]interface{}{
+					"foo": "bar",
+				},
+				b: map[string]interface{}{
+					"foo": "bar",
+					"baz": "qux",
+				},
+			},
+			expected: "+ baz: qux\n  foo: bar\n",
+		},
+		{
+			name: "New non-string elements should be prefixed with a plus sign",
+			args: args{
+				a: map[string]interface{}{
+					"foo": "bar",
+				},
+				b: map[string]interface{}{
+					"foo": "bar",
+					"baz": []string{"qux"},
+					"qux": map[string]string{
+						"quux": "corge",
+					},
+				},
+			},
+			expected: "+ baz:\n+ - qux\n  foo: bar\n+ qux:\n+   quux: corge\n",
+		},
```

Comment:
Maybe we could include a simple string removal case and a deep map removal case as analogs to the new elements cases? wdyt?

Reply:
100%